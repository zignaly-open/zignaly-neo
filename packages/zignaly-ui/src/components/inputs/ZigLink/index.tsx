import React from "react";
import { styled, LinkProps, Link } from "@mui/material";

const ZigLink = styled(
  ({ rel, target = "_blank", underline = "hover", ...props }: LinkProps & {}) => (
    <Link underline={underline} rel={rel || "noopener noreferrer"} target={target} {...props} />
  ),
)``;

export default ZigLink as typeof Link;
