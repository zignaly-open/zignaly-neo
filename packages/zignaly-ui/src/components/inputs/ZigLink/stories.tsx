import React, { useState } from "react";
import { Meta, Story } from "@storybook/react";
import ZigLink from "./index";
import { ButtonGroup, ButtonGroupProps } from "@mui/material";
import { Add } from "@mui/icons-material";
import type * as All from "@mui/material/node_modules/@mui/system/styleFunctionSx";
import ZigTypography from "../../display/ZigTypography";

export default {
  title: "Inputs/ZigLink",
  component: ZigLink,
  argTypes: {
    href: {
      type: "string",
      defaultValue: "https://xfuturum.com",
    },
  },
} as Meta<typeof ZigLink>;

const Template: Story<typeof ZigLink> = (args) => <ZigLink {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: "Amount to Withdraw",
};

const TemplateTextButton: Story<typeof ZigLink> = (args) => {
  return (
    <>
      <ZigTypography variant={"h1"}>
        Hello <ZigLink {...args} />
      </ZigTypography>
      <ZigTypography variant={"h2"}>
        Hello <ZigLink {...args} />
      </ZigTypography>
      <ZigTypography variant={"h3"}>
        Hello <ZigLink {...args} />
      </ZigTypography>
      <ZigTypography variant={"h4"}>
        Hello <ZigLink {...args} />
      </ZigTypography>
      <ZigTypography variant={"body1"} component={"p"}>
        Hello, let me tell you a story of a button that lived in a very very long text. Hello, let
        me tell you a story of a button that lived in a very very long text. Hello, let me tell you
        a story of a button that lived in a very very long text. Yes, an <ZigLink {...args} /> lived
        in a very long text and then died unhappily ever after. Yes, that was story of a button that
        lived in a very very long text.
      </ZigTypography>

      <ZigTypography variant={"body1"} component={"p"}>
        Hello, let me tell you a story of a button that lived in a very very long text. Hello, let
        me tell you a story of a button that lived in a very very long text. Yes, a{" "}
        <ZigLink href="https://xfuturum.com">Button</ZigLink> can be a link, imagine that.
      </ZigTypography>

      <ZigTypography variant={"body2"} component={"p"}>
        Hello <ZigLink {...args} /> button in text in a very long
      </ZigTypography>
    </>
  );
};

const buttonExampleArgs = {
  children: "Amount to Withdraw",
  href: "https://xfuturum.com",
};

export const TextButton = TemplateTextButton.bind({});
TextButton.args = buttonExampleArgs;
