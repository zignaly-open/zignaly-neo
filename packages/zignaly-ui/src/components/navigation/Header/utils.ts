// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
