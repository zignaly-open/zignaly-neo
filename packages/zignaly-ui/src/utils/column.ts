import styled from "styled-components";

export const Column = styled.div<{
  gap?: number;
  justifyContent?: "center" | "end" | "start" | "space-between";
  alignItems?: "center" | "end" | "start" | "space-between";
  textAlign?: "end" | "center" | "start";
  height?: string;
  flex?: number;
  width?: string;
}>`
  display: flex;
  flex-direction: column;
  flex: ${(props) => props.flex};
  gap: ${(props) => props.gap}px;
  justify-content: ${(props) => props.justifyContent};
  align-items: ${(props) => props.alignItems};
  text-align: ${(props) => `${props.textAlign}`};
  height: ${(props) => `${props.height}`};
  width: ${(props) => `${props.width}`};
`;
