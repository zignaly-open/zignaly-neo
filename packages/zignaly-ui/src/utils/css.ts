import { css } from "styled-components";

export const NiceScrollbar = css`
  &::-webkit-scrollbar {
    width: 6px;
  }

  /* Track */
  &::-webkit-scrollbar-track {
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
  }

  /* Handle */
  &::-webkit-scrollbar-thumb {
    background: rgba(255, 255, 255, 0.1);
    border-radius: 12px;
  }

  /* Handle on hover */
  &::-webkit-scrollbar-thumb:hover {
    background: rgba(255, 255, 255, 0.15);
  }
`;
