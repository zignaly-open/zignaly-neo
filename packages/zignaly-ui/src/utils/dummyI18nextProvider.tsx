import i18n from "i18next";
import React from "react";
import { I18nextProvider, initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
  lng: "en",
  fallbackLng: "en",
  ns: ["common"],
  defaultNS: "common",
  interpolation: {
    escapeValue: false,
  },
  resources: { en: { common: { xfuturum: "The technical partner you can rely on" } } },
});

export const storybookI18nextDecorator = (story: () => JSX.Element) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{story()}</I18nextProvider>
);

export const JestI18nextDecorator: React.FC<{ children: JSX.Element }> = ({ children }) => (
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
);
