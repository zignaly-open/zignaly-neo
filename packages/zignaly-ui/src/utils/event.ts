export const changeEvent = (name: string | undefined, value: any) => {
  return {
    target: {
      value,
      name,
    },
    type: "change",
  };
};
