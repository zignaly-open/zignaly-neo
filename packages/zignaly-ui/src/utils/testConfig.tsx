import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
