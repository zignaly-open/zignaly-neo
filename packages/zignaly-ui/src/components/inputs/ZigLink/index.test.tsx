import React from "react";
import ZigLink from ".";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, waitFor } from "@testing-library/react";

describe("components/inputs/ZigLink", () => {
  describe("rendering", () => {
    it("renders with default props", async () => {
      const { container } = renderWithProvidersUi(<ZigLink href={"#"} id={"link-test"} />);

      const link = container.querySelector("#link-test") as Element;
      expect(link).toBeVisible();
      const linkStyles = getComputedStyle(link);
      expect(link).toMatchSnapshot();
      expect(linkStyles).toMatchSnapshot();
    });
  });

  describe("actions", () => {
    it("should handle click", async () => {
      const handleClick = jest.fn();
      const { container } = renderWithProvidersUi(
        <ZigLink href={"#"} id={"link-test"} onClick={handleClick} />,
      );

      const link = container.querySelector("#link-test") as Element;
      fireEvent.click(link);
      await waitFor(() => {
        expect(handleClick).toHaveBeenCalledTimes(1);
      });
    });
  });
});
