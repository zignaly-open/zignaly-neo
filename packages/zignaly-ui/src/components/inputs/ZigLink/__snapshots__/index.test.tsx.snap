// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inputs/ZigLink renders with default props 1`] = `
<a
  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-94cnhm-MuiTypography-root-MuiLink-root"
  href="#"
  id="link-test"
  rel="noopener noreferrer"
  target="_blank"
/>
`;

exports[`components/inputs/ZigLink renders with default props 2`] = `
CSSStyleDeclaration {
  "0": "margin",
  "1": "letter-spacing",
  "2": "color",
  "3": "text-decoration",
  "4": "cursor",
  "5": "visibility",
  "_importants": {
    "color": "",
    "cursor": "",
    "font": "",
    "letter-spacing": "",
    "margin": "",
    "text-decoration": "",
    "visibility": undefined,
  },
  "_length": 6,
  "_onChange": [Function],
  "_values": {
    "color": "rgb(38, 196, 193)",
    "cursor": "pointer",
    "letter-spacing": "0.55px",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "text-decoration": "none",
    "visibility": "visible",
  },
}
`;
